### PR TITLE
Accept subprotocols in connect action

### PR DESCRIPTION
**Pull request by Eddie-CooRo**
related to #73

### CODE REVIEW

File: src/ReduxWebSocket.ts [hunk 1]
```diff
@@ -66,8 +66,8 @@ export default class ReduxWebSocket {
     const { prefix } = this.options;
 
     this.lastSocketUrl = payload.url;
-    this.websocket = new WebSocket(payload.url);
-
+    this.websocket = new WebSocket(payload.url, payload.subprotocols);
```
Comment (brianmcallister) [inline src/ReduxWebSocket.ts]: Could we change the name of this option to `protocols`? That's what it's called in the spec. See: https://html.spec.whatwg.org/multipage/web-sockets.html#the-websocket-interface

Comment (Eddie-CooRo) [reply src/ReduxWebSocket.ts]: Yeah. of course

**Comment by brianmcallister** [comment]
I'm not too sure what's up with the failing tests right now. I think it's a problem with permissions on Semaphore. I might actually just switch CI providers if I can't figure it out.

In the meantime, it would be great if a test could be added for the new option as well.

**Comment by Eddie-CooRo** [comment]
Is it enough to add a subprotocol field to the default action in the connect's describe block?

**Comment by brianmcallister** [comment]
I was more thinking a separate `it` block, and then assert that the `protocols` key is optional. Related, what I would like to do in the future (see #60) is set up the types so that all of the action payloads are type safe as well.

**Comment by Eddie-CooRo** [comment]
Ok. I don't get what should I do with tests; there is a `beforeEach` block in the `ReduxWebSocket`s describe block which creates a new `ReduxWebSocket` for each test, And probably I should add the payload field here and in the `connect`s describe block to make sure that `global.WebSocket` has been called with URL and the payload as the input parameters. But now how can I assert that the protocol's key is optional? Maybe I should create a separate `beforeEach` and a separate `describe` block for the new test?

**Comment by brianmcallister** [comment]
I don't actually know if there's a way for me to push code directly to your fork, but here's what I'm thinking:

```js
    it('optionally takes a protocols option', () => {
      const connectAction = { type: 'SEND', payload: { url, protocols: 'test' } };

      global.WebSocket.mockClear();

      reduxWebSocket.connect(store, connectAction as Action);

      expect(global.WebSocket).toHaveBeenCalledTimes(1);
      expect(global.WebSocket).toHaveBeenCalledWith(url, 'test');
    });
```

So that tests that you can dispatch `protocols` in the connection action, and then you can change the assertion just above to:

```js
expect(global.WebSocket).toHaveBeenCalledWith(url, undefined);
```

...because now, we're always passing in a second argument, but it's going to be `undefined` if you never set `protocols`.

**Comment by Eddie-CooRo** [comment]
Hi @brianmcallister.
Can you please take a look at the new commits?

**Comment by Eddie-CooRo** [comment]
Any update?

**Comment by brianmcallister** [comment]
@Eddie-CooRo I just merged #77. Please let me know if that works for your needs. Thanks!

**Comment by brianmcallister** [comment]
@Eddie-CooRo I'm going to go ahead and close this, but feel free to open another issue if you'd like to discuss this further. Thanks for the input!

**Pull request closed without merging**